packages/flutter_driver/lib/src/driver.dart

```diff
@@ -0,0 +1,166 @@
+    FlutterDriver driver = newFlutterDriver(client, isolate);
+
+    // Attempts to resume the isolate, but does not crash if it fails because
+    // the isolate is already resumed. There could be a race with other tools,
+    // such as a debugger, any of which could have resumed the isolate.
+    Future resumeLeniently() {
+      _log.trace('Attempting to resume isolate');
+      return isolate.resume().catchError((e) {
+        const vmMustBePausedCode = 101;
+        if (e is rpc.RpcException && e.code == vmMustBePausedCode) {
+          // No biggie; something else must have resumed the isolate
+          _log.warning(
+            'Attempted to resume an already resumed isolate. This may happen '
+            'when we lose a race with another tool (usually a debugger) that '
+            'is connected to the same isolate.');
+        } else {
+          // Failed to resume due to another reason. Fail hard.
+          throw e;
+        }
+      });
+    }
+
+    // Attempt to resume isolate if it was paused
+    if (isolate.pauseEvent is VMPauseStartEvent) {
+      _log.trace('Isolate is paused at start.');
+
+      // Waits for a signal from the VM service that the extension is registered
+      Future waitForServiceExtension() {
+        return isolate.onServiceExtensionAdded.firstWhere((VMServiceExtension ext) {
+          return ext.method == _flutterExtensionMethod;
+        });
+      }
+
+      // If the isolate is paused at the start, e.g. via the --start-paused
+      // option, then the VM service extension is not registered yet. Wait for
+      // it to be registered.
+      const timeout = const Object();
+      Future whenResumed = resumeLeniently();
+      Future whenServiceExtensionReady = Future.any([
+        waitForServiceExtension(),
+        // We will never receive the extension event if the user does not
+        // register it. If that happens time out.
+        new Future.delayed(const Duration(seconds: 10), () => timeout)
+      ]);
+      await whenResumed;
+      _log.trace('Waiting for service extension');
+      dynamic signal = await whenServiceExtensionReady;
+      if (signal == timeout) {
+        throw new DriverError(
+          'Timed out waiting for Flutter Driver extension to become available. '
+          'To enable the driver extension call registerFlutterDriverExtension '
+          'first thing in the main method of your appilcation.');
+      }
+    } else if (isolate.pauseEvent is VMPauseExitEvent ||
+        isolate.pauseEvent is VMPauseBreakpointEvent ||
+        isolate.pauseEvent is VMPauseExceptionEvent ||
+        isolate.pauseEvent is VMPauseInterruptedEvent) {
+      // If the isolate is paused for any other reason, assume the extension is
+      // already there.
+      _log.trace('Isolate is paused mid-flight.');
+      await resumeLeniently();
+    } else {
+      _log.trace('Isolate is not paused. Assuming application is ready.');
```

Comment:
have an if branch for VMResumeEvent and then an else branch for "unknown event type ${isolate.pauseEvent.runtimeType}"  or something.